src/js/express.js

```diff
@@ -0,0 +1,179 @@
+
+                // Recurse from here on down.
+                that.connectDirectDescendants(grandchildComponent, childPath + "." + grandchildNickname);
+            }
+            else if (fluid.hasGrade(grandchildComponent.options, "gpii.express.middleware")) {
+                if (childComponent.options.router) {
+                    // We have to wire our children in with our path to preserve relative pathing, so that their router will begin with our path.
+                    childComponent.options.router.use(childComponent.options.path, grandchildComponent.getMiddlewareFunction());
+                }
+                else {
+                    throw(new Error({message: "A component must expose a router in order to work with child middleware components."}));
+                }
+            }
+        }
+    }
+
+    // After our child components are in place, wire the router to itself.
+    //
+    // This must be done here because we want to give children the chance to own part of the path before we take the rest over.
+    //
+    // The path and method have to be used here so that parameters will be parsed correctly.
+    if (fluid.hasGrade(childComponent.options, "gpii.express.router")) {
+        childComponent.options.router[childComponent.options.method](childComponent.options.path, childComponent.getRouterFunction());
+    }
+};
+
+gpii.express.init = function(that) {
+    if (!that.options.config || !that.options.config.express) {
+        console.error("Cannot initialize express because you have not supplied a 'config' option.");
+        return;
+    }
+
+    var express  = require("express");
+    that.express = express();
+
+    // Wire together all routers and components, beginning with ourselves
+    for (var a = 0; a < that.options.members.directChildrenOfInterest.length; a++) {
+        var directChildNickname = that.options.members.directChildrenOfInterest[a];
+        var childComponent      = that[directChildNickname];
+        if (fluid.hasGrade(childComponent.options, "gpii.express.router")) {
+            // The router has to wire its own paths to preserve methods and path variables.
+            // We just "use" it at the root level, and let it do the rest.
+            that.express.use("/", childComponent.options.router);
+        }
+        else if (fluid.hasGrade(childComponent.options, "gpii.express.middleware")) {
+            that.express.use(childComponent.getMiddlewareFunction());
+        }
+
+        // Recurse from here on down.
+        that.connectDirectDescendants(childComponent, directChildNickname);
+    }
+
+    that.express.set("port", that.options.config.express.port);
+    that.server = that.express.listen(that.options.config.express.port, function(){
+        console.log("Express server listening on port " + that.express.get("port"));
+
+        console.log("Express started...");
+        that.events.started.fire(that.express);
```

Comment:
I updated the names for both the `stopped` and `started` events.

I also added `that` as a second argument when firing the event.